### PR TITLE
feat: allow system package installation for conan by default

### PR DESCRIPTION
**Pull request by FeignClaims**
This makes conan in cmake more transparent.

### CODE REVIEW

**Review by aminya — APPROVED** [review]

**Comment by aminya** [comment]
Does this work with Conan 1 as well as 2?

**Comment by FeignClaims** [comment]
> Does this work with Conan 1 as well as 2?

Only conan 2. Currently [cmake-conan integration for Conan 1.X is no longer mantainedand](https://github.com/conan-io/cmake-conan/tree/develop) and [Conan Center has stopped receiving updates for Conan 1.x packages](https://blog.conan.io/2024/09/30/Conan-Center-will-stop-receiving-updates-for-Conan-1.html), so I didn't consider how to enable it for conan 1.

**Comment by aminya** [comment]
> > Does this work with Conan 1 as well as 2?
> 
> Only conan 2. Currently [cmake-conan integration for Conan 1.X is no longer mantainedand](https://github.com/conan-io/cmake-conan/tree/develop?rgh-link-date=2024-11-10T06%3A18%3A18Z) and [Conan Center has stopped receiving updates for Conan 1.x packages](https://blog.conan.io/2024/09/30/Conan-Center-will-stop-receiving-updates-for-Conan-1.html), so I didn't consider how to enable it for conan 1.

Could you conditionally add the flags for Conan 2?

File: src/Conan.cmake [hunk 1]
```diff
@@ -217,7 +217,11 @@ macro(_run_conan2)
   endif()
 
   if(NOT _args_INSTALL_ARGS)
-    set(_args_INSTALL_ARGS "--build=missing")
+    set(_args_INSTALL_ARGS
+        --build=missing
+        -c tools.system.package_manager:mode=install
+        -c tools.system.package_manager:sudo=True
+    )
```
Comment (aminya) [inline src/Conan.cmake:224]: > > Does this work with Conan 1 as well as 2?
> 
> Only conan 2. Currently [cmake-conan integration for Conan 1.X is no longer mantainedand](https://github.com/conan-io/cmake-conan/tree/develop?rgh-link-date=2024-11-10T06%3A18%3A18Z) and [Conan Center has stopped receiving updates for Conan 1.x packages](https://blog.conan.io/2024/09/30/Conan-Center-will-stop-receiving-updates-for-Conan-1.html), so I didn't consider how to enable it for conan 1.

Could you conditionally add the flags for Conan 2?

**Review by aminya — CHANGES_REQUESTED** [review]

File: src/Conan.cmake (hunk 1 above)
Comment (FeignClaims) [reply src/Conan.cmake:224]: It's not necessary. The function `_run_conan2` here is just for conan 2. The user interface `run_conan` checks user's conan version and dispatch the call to either `_run_conan2` or `_run_conan1`

**Comment by aminya** [comment]
Apparently, it's supported for Conan 1 as well
https://docs.conan.io/1/reference/conanfile/tools/system/package_manager.html#conan-tools-system-package-manager

**Comment by FeignClaims** [comment]
> Apparently, it's supported for Conan 1 as well https://docs.conan.io/1/reference/conanfile/tools/system/package_manager.html#conan-tools-system-package-manager

done

**Review by aminya — APPROVED** [review]

**Comment by aminya** [comment]
Thank you!

**Pull request merged**